### PR TITLE
Now linking the application secrets and service account to the CR instead.

### PR DESCRIPTION
**Pull request by olofwalker**

### CODE REVIEW

File: kubectl-cloudflow/deploy/deploy.go [hunk 1]
```diff
@@ -350,16 +360,15 @@ func findSecretName(spec *domain.CloudflowApplicationSpec, streamletName string)
...
 	labels["com.lightbend.cloudflow/streamlet-name"] = name
 	secret := &corev1.Secret{
 		Type: corev1.SecretTypeOpaque,
 		ObjectMeta: metav1.ObjectMeta{
-			Name:            name,
-			Namespace:       appID,
-			Labels:          labels,
-			OwnerReferences: []metav1.OwnerReference{cloudflowOperatorOwnerReference},
+			Name:      name,
+			Namespace: appID,
```
Comment (skonto) [inline kubectl-cloudflow/deploy/deploy.go]: appID has a value of a namespace?

Comment (debasishg) [reply kubectl-cloudflow/deploy/deploy.go]: we deploy apps in the namespace = appID ..

Comment (skonto) [reply kubectl-cloudflow/deploy/deploy.go]: oh yes right.

File: kubectl-cloudflow/domain/cloudflowapplication.go [hunk 2]
```diff
@@ -5,7 +5,7 @@ import (
 	"fmt"
 
 	v1 "k8s.io/api/core/v1"
-	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (skonto) [inline kubectl-cloudflow/domain/cloudflowapplication.go]: could you fix this also in some other places like the docker unit tests?

File: kubectl-cloudflow/deploy/deploy.go [hunk 3]
```diff
@@ -336,11 +336,21 @@ func CreateSecretsData(spec *domain.CloudflowApplicationSpec, configurationKeyVa
...
 		secretName := findSecretName(spec, streamlet.Name)
-		streamletSecretNameMap[secretName] = createSecret(spec.AppID, secretName, secretMap, cloudflowOperatorOwnerReference)
+		streamletSecretNameMap[secretName] = createSecret(spec.AppID, secretName, secretMap)
 	}
 	return streamletSecretNameMap
 }
 
+// UpdateSecretsWithOwnerReference updates the secret with the ownerreference passed in
+func UpdateSecretsWithOwnerReference(cloudflowCROwnerReference metav1.OwnerReference, secrets map[string]*corev1.Secret) map[string]*corev1.Secret {
+
```
Comment (skonto) [inline kubectl-cloudflow/deploy/deploy.go]: nit:remove space?

File: kubectl-cloudflow/cmd/configure.go [hunk 4]
```diff
@@ -66,9 +66,11 @@ func (c *configureApplicationCMD) configureImpl(cmd *cobra.Command, args []strin
 	}
 
 	// Get the Cloudflow operator ownerReference
-	ownerReference := version.GetOwnerReferenceForCloudflowOperator()
+	ownerReference := applicationCR.GenerateOwnerReference()
+
+	streamletNameSecretMap := deploy.CreateSecretsData(&applicationCR.Spec, configurationKeyValues)
+	streamletNameSecretMap = deploy.UpdateSecretsWithOwnerReference(ownerReference, streamletNameSecretMap)
```
Comment (skonto) [inline kubectl-cloudflow/cmd/configure.go]: If we delete the app CR, all resources are removed rigjt? or is anything that is pending to complete this fix?

**Comment by skonto** [comment]
@olofwalker could you also update the description of the PR?

File: kubectl-cloudflow/domain/cloudflowapplication_test.go [hunk 5]
```diff
@@ -47,6 +47,21 @@ func Test_validateCloudflowApplicationSpec(t *testing.T) {
...
+	jsonError := json.Unmarshal([]byte(applicationConfiguration), &app)
+	assert.Empty(t, jsonError)
+
+	cr := NewCloudflowApplication(app)
+
+	ownerReference := cr.GenerateOwnerReference()
+	assert.True(t, ownerReference.Kind == "CloudflowApplication")
+	assert.True(t, ownerReference.APIVersion == "cloudflow.lightbend.com/v1alpha1")
+	assert.True(t, ownerReference.Name == "sensor-data-scala")
+
```
Comment (skonto) [inline kubectl-cloudflow/domain/cloudflowapplication_test.go]: could you also test about the other fields like uid etc?

**Comment by skonto** [comment]
Minor comments, I have not run the PR, but LGTM.

File: kubectl-cloudflow/domain/cloudflowapplication_test.go (hunk 5 above)
Comment (olofwalker) [reply kubectl-cloudflow/domain/cloudflowapplication_test.go]: `uid` is not relevant, its set by k8s and will be present when you fetch the CR from the cluster, but not in a case like this.

File: kubectl-cloudflow/cmd/configure.go (hunk 4 above)
Comment (olofwalker) [reply kubectl-cloudflow/cmd/configure.go]: That is the intention, the secrets was previously tied to the operator, but that was more of a hack.

File: kubectl-cloudflow/domain/cloudflowapplication.go (hunk 2 above)
Comment (olofwalker) [reply kubectl-cloudflow/domain/cloudflowapplication.go]: Fixed.

File: kubectl-cloudflow/deploy/deploy.go (hunk 3 above)
Comment (olofwalker) [reply kubectl-cloudflow/deploy/deploy.go]: sure :)

**Pull request merged**